examples/shor.py

```diff
@@ -104,8 +108,10 @@ def high_level_gates(eng, cmd):
 if __name__ == "__main__":
     # build compilation engine list
     resource_counter = ResourceCounter()
-    compilerengines = [AutoReplacer(), InstructionFilter(high_level_gates),
-                       TagRemover(), LocalOptimizer(3), AutoReplacer(),
+    dh = DecompositionRuleSet(modules=[projectq.libs.math,
```

Comment:
Also, we'll need decomposition rules in the jupyter notebook (in examples/) or, alternatively
```
compilerengines = projectq.default_engines() + [CommandPrinter()]
```
(plus an `import projectq` somewhere).